types.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2025 Pinterest
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package thriftcheck
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"go.uber.org/thriftrw/ast"
+)
+
+// TypeMatcher represents a way to match against AST nodes.
+type TypeMatcher interface {
+	Matches(c *C, n ast.Node) bool
+	Name() string
+}
```

Comment:
`TypeMatcher` is basically internal now that we use `ThriftType` outside of this file.

And specifically, `Name()` is no longer needed and can be removed.